### ISSUE TITLE
[BUG] Page Indexer shows errors while strict mode is set and not all pages are translated

### ISSUE THREAD

**Issue by PhilippKuhlmay** [TYPO3-Solr/ext-solr#2637]
**Describe the bug**
Page-indexer shows error with non-translated pages while strict mode is set.

**To Reproduce**
1. Create two or more languages within the TYPO3 Installation. Enable strict mode for language mode.
2. Create a page with the default language
3. Don't create a translation for this page.
4. Put "pages" in the indexer queue and start indexing the queue

**Expected behavior**
All available pages should be indexed, but missing translations shouldn't be indexed.

**Actual behavior**
The indexer tries to index the not-translated pages and shows an error.

**Screenshots**
![Bildschirmfoto 2020-06-08 um 11 19 41](https://user-images.githubusercontent.com/57705869/84014032-039dd100-a97a-11ea-8a32-e6cdc21808dd.png)

**Code**
`404: GuzzleHttp\Exception\RequestException: cURL error 56: Illegal or missing hexadecimal sequence in chunked-encoding (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) in /.../vendor/guzzlehttp/guzzle/src/Handler/CurlFactory.php:201
`

**Used versions (please complete the following information):**
 - TYPO3 Version: 10.4.3
 - Browser: Chrome
 - EXT:solr Version: 11.0.1
 - Used Apache Solr Version: 8.5.2
 - PHP Version: 7.4.4
 - MySQL Version: 5.5.5

**Comment by websi** [comment]
In the current version 11.0.2 and an configured page not found handling, the scheduler breaks with the following error message:

```
Core: Exception handler (CLI): Uncaught TYPO3 Exception: Call to a member function getUserInfo() on null | Error thrown in file public/typo3/sysext/core/Classes/Error/PageErrorHandler/PageContentErrorHandler.php in line 149

```

Error handling configuration from site:

```
errorHandling:
  -
    errorCode: '404'
    errorHandler: Page
    errorContentSource: 't3://page?uid=190'
```

The TYPO3 core try to reuse information from current url, but the current url does not exist at cli context

**Comment by websi** [comment]
The problem not only exist for the page indexer. Also records on an non translated storage pages.
Workaround: Translate the storage pages.

Technical: Every call of the method `\ApacheSolrForTypo3\Solr\FrontendEnvironment\Tsfe::initializeTsfe` with an an non existing`$language` for `$pageId`

**Comment by mvoehringer** [comment]
Is there any known workaround? In my project I can index news because it always returns an error
`Uncaught TYPO3 Exception: Call to a member function getUserInfo() on null | Error thrown in file /var/www/html/private/typo3/sysext/core/Classes/Error/PageErrorHandler/PageContentErrorHandler.php in line 149. `

And if I disable the 404 error handling in the site setting I see the error:
`1533931388: TYPO3\CMS\Core\Http\ImmediateResponseException in /var/www/html/private/typo3/sysext/frontend/Classes/Controller/TypoScriptFrontendController.php:2226
Stack trace:
#0 /var/www/html/private/typo3conf/ext/solr/Classes/FrontendEnvironment/Tsfe.php(114): TYPO3\CMS\Frontend\Controller\TypoScriptFrontendController->settingLanguage()
#1 /var/www/html/private/typo3conf/ext/solr/Classes/FrontendEnvironment.php(47): ApacheSolrForTypo3\Solr\FrontendEnvironment\Tsfe->initializeTsfe(1, 1)
#2 /var/www/html/private/typo3conf/ext/solr/Classes/IndexQueue/Indexer.php(358): ApacheSolrForTypo3\Solr\FrontendEnvironment->initializeTsfe(1, 1)
#3 /var/www/html/private/typo3conf/ext/solr/Classes/IndexQueue/Indexer.php(177): ApacheSolrForTypo3\Solr\IndexQueue\Indexer->itemToDocument(Object(ApacheSolrForTypo3\Solr\IndexQueue\Item), 1)
#4 /var/www/html/private/typo3conf/ext/solr/Classes/IndexQueue/Indexer.php(149): ApacheSolrForTypo3\Solr\IndexQueue\Indexer->indexItem(Object(ApacheSolrForTypo3\Solr\IndexQueue\Item), 1)
#5 /var/www/html/private/typo3conf/ext/solr/Classes/Domain/Index/IndexService.php(205): ApacheSolrForTypo3\Solr\IndexQueue\Indexer->index(Object(ApacheSolrForTypo3\Solr\IndexQueue\Item))
#6 /var/www/html/private/typo3conf/ext/solr/Classes/Domain/Index/IndexService.php(131): ApacheSolrForTypo3\Solr\Domain\Index\IndexService->indexItem(Object(ApacheSolrForTypo3\Solr\IndexQueue\Item), Object(ApacheSolrForTypo3\Solr\System\Configuration\TypoScriptConfiguration))
#7 /var/www/html/private/typo3conf/ext/solr/Classes/Controller/Backend/Search/IndexQueueModuleController.php(259): ApacheSolrForTypo3\Solr\Domain\Index\IndexService->indexItems(10)
#8 /var/www/html/private/typo3/sysext/extbase/Classes/Mvc/Controller/ActionController.php(370): ApacheSolrForTypo3\Solr\Controller\Backend\Search\IndexQueueModuleController->doIndexingRunAction()
#9 /var/www/html/private/typo3/sysext/extbase/Classes/Mvc/Controller/ActionController.php(213): TYPO3\CMS\Extbase\Mvc\Controller\ActionController->callActionMethod()
#10 /var/www/html/private/typo3/sysext/extbase/Classes/Mvc/Dispatcher.php(89): TYPO3\CMS\Extbase\Mvc\Controller\ActionController->processRequest(Object(TYPO3\CMS\Extbase\Mvc\Web\Request), Object(TYPO3\CMS\Extbase\Mvc\Web\Response))
#11 /var/www/html/private/typo3/sysext/extbase/Classes/Mvc/Web/BackendRequestHandler.php(36): TYPO3\CMS\Extbase\Mvc\Dispatcher->dispatch(Object(TYPO3\CMS\Extbase\Mvc\Web\Request), Object(TYPO3\CMS\Extbase\Mvc\Web\Response))
#12 /var/www/html/private/typo3/sysext/extbase/Classes/Core/Bootstrap.php(226): TYPO3\CMS\Extbase\Mvc\Web\BackendRequestHandler->handleRequest()
#13 [internal function]: TYPO3\CMS\Extbase\Core\Bootstrap->handleBackendRequest(Object(TYPO3\CMS\Core\Http\ServerRequest))
#14 /var/www/html/private/typo3/sysext/backend/Classes/Http/RouteDispatcher.php(66): call_user_func_array(Array, Array)
#15 /var/www/html/private/typo3/sysext/backend/Classes/Http/RequestHandler.php(94): TYPO3\CMS\Backend\Http\RouteDispatcher->dispatch(Object(TYPO3\CMS\Core\Http\ServerRequest))
#16 /var/www/html/private/typo3/sysext/extbase/Classes/Middleware/SignalSlotDeprecator.php(49): TYPO3\CMS\Backend\Http\RequestHandler->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#17 /var/www/html/private/typo3/sysext/core/Classes/Http/MiddlewareDispatcher.php(172): TYPO3\CMS\Extbase\Middleware\SignalSlotDeprecator->process(Object(TYPO3\CMS\Core\Http\ServerRequest), Object(TYPO3\CMS\Backend\Http\RequestHandler))
#18 /var/www/html/private/typo3/sysext/backend/Classes/Middleware/SiteResolver.php(69): class@anonymous->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#19 /var/www/html/private/typo3/sysext/core/Classes/Http/MiddlewareDispatcher.php(172): TYPO3\CMS\Backend\Middleware\SiteResolver->process(Object(TYPO3\CMS\Core\Http\ServerRequest), Object(class@anonymous))
#20 /var/www/html/private/typo3/sysext/backend/Classes/Middleware/AdditionalResponseHeaders.php(41): class@anonymous->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#21 /var/www/html/private/typo3/sysext/core/Classes/Http/MiddlewareDispatcher.php(172): TYPO3\CMS\Backend\Middleware\AdditionalResponseHeaders->process(Object(TYPO3\CMS\Core\Http\ServerRequest), Object(class@anonymous))
#22 /var/www/html/private/typo3/sysext/backend/Classes/Middleware/OutputCompression.php(47): class@anonymous->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#23 /var/www/html/private/typo3/sysext/core/Classes/Http/MiddlewareDispatcher.php(172): TYPO3\CMS\Backend\Middleware\OutputCompression->process(Object(TYPO3\CMS\Core\Http\ServerRequest), Object(class@anonymous))
#24 /var/www/html/private/typo3/sysext/backend/Classes/Middleware/BackendUserAuthenticator.php(77): class@anonymous->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#25 /var/www/html/private/typo3/sysext/core/Classes/Http/MiddlewareDispatcher.php(172): TYPO3\CMS\Backend\Middleware\BackendUserAuthenticator->process(Object(TYPO3\CMS\Core\Http\ServerRequest), Object(class@anonymous))
#26 /var/www/html/private/typo3/sysext/backend/Classes/Middleware/BackendRouteInitialization.php(58): class@anonymous->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#27 /var/www/html/private/typo3/sysext/core/Classes/Http/MiddlewareDispatcher.php(172): TYPO3\CMS\Backend\Middleware\BackendRouteInitialization->process(Object(TYPO3\CMS\Core\Http\ServerRequest), Object(class@anonymous))
#28 /var/www/html/private/typo3/sysext/backend/Classes/Middleware/ForcedHttpsBackendRedirector.php(55): class@anonymous->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#29 /var/www/html/private/typo3/sysext/core/Classes/Http/MiddlewareDispatcher.php(172): TYPO3\CMS\Backend\Middleware\ForcedHttpsBackendRedirector->process(Object(TYPO3\CMS\Core\Http\ServerRequest), Object(class@anonymous))
#30 /var/www/html/private/typo3/sysext/backend/Classes/Middleware/LockedBackendGuard.php(73): class@anonymous->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#31 /var/www/html/private/typo3/sysext/core/Classes/Http/MiddlewareDispatcher.php(172): TYPO3\CMS\Backend\Middleware\LockedBackendGuard->process(Object(TYPO3\CMS\Core\Http\ServerRequest), Object(class@anonymous))
#32 /var/www/html/private/typo3/sysext/core/Classes/Middleware/NormalizedParamsAttribute.php(45): class@anonymous->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#33 /var/www/html/private/typo3/sysext/core/Classes/Http/MiddlewareDispatcher.php(172): TYPO3\CMS\Core\Middleware\NormalizedParamsAttribute->process(Object(TYPO3\CMS\Core\Http\ServerRequest), Object(class@anonymous))
#34 /var/www/html/private/typo3/sysext/core/Classes/Http/MiddlewareDispatcher.php(78): class@anonymous->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#35 /var/www/html/private/typo3/sysext/core/Classes/Http/AbstractApplication.php(85): TYPO3\CMS\Core\Http\MiddlewareDispatcher->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#36 /var/www/html/private/typo3/sysext/backend/Classes/Http/Application.php(63): TYPO3\CMS\Core\Http\AbstractApplication->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#37 /var/www/html/private/typo3/sysext/core/Classes/Http/AbstractApplication.php(97): TYPO3\CMS\Backend\Http\Application->handle(Object(TYPO3\CMS\Core\Http\ServerRequest))
#38 /var/www/html/public/typo3/index.php(26): TYPO3\CMS\Core\Http\AbstractApplication->run()
#39 /var/www/html/public/typo3/index.php(27): {closure}()
#40 {main}`

**Comment by dkd-kaehm** [comment]
We can fix that issue on EXT:solr 11.5.x only, because of complexity and deep interlacement of TSFE usages and initialization in previous versions.

Unfortunately this issue was not found on searching issues which must be closed within the EPIC task of TSFE refactoring.

I'll close that issue manually.

### Fixed in "!!![TASK] Improve and Fix TSFE Initialization" https://github.com/TYPO3-Solr/ext-solr/commit/a246cb8e34d80da6227445c0c800620f16264a00
